CONTRIBUTING_MODELS.md

```diff
@@ -72,168 +119,174 @@ contribution! 🙂
-1) converts the original preset weights to our format, and
-2) checks whether the outputs of the original model and your implemented model are close enough.
+##### Unit Tests
 
-It is okay if you demonstrate it for one preset at this stage; you can do the conversion
-for the other presets when you officially add presets to the library at a later stage.
+- Test basic usage: instantiation, forward pass with dummy data from the 
+  preprocessor, and model compilation.
 
-#### Add Unit Tests
+---
 
-It is essential to add units tests. These unit tests are basic and mostly check
-whether the forward pass goes through successfully, whether the model can be saved
-and loaded correctly, etc.
+### Step 5: PR #4 – Presets and End-to-End Validation
 
-### Step 3: PR #2 - Add XXTokenizer
+After PRs 1–3 are merged, create:
 
-#### Tokenizer
+#### Preset Configuration
 
-Most text models nowadays use subword tokenizers such as WordPiece, SentencePiece
-and BPE Tokenizer. Since KerasHub has implementations of most of the popular
-subword tokenizers, the model tokenizer layer typically inherits from a base
-tokenizer class.
+- Add `xx_presets.py`
+- Include model args, checkpoint URLs, vocabulary paths
 
-For example, DistilBERT uses the WordPiece tokenizer. So, we can introduce a new
-class, `DistilBertTokenizer`, which inherits from `keras_hub.tokenizers.WordPieceTokenizer`.
-All the underlying actual tokenization will be taken care of by the superclass.
+Use the [Kaggle org page](https://www.kaggle.com/organizations/kerashub/models) 
+to stage and test.
 
-The important thing here is adding "special tokens". Most models have
-special tokens such as beginning-of-sequence token, end-of-sequence token,
-mask token, pad token, etc. These have to be
-[added as member attributes](https://github.com/keras-team/keras-hub/blob/master/keras_hub/src/models/distil_bert/distil_bert_tokenizer.py#L91-L105)
-to the tokenizer class. These member attributes are then accessed by the
-preprocessor layers.
+#### `from_preset()` Functions
 
-For a full list of the tokenizers KerasHub offers, please visit
-[this link](https://keras.io/api/keras_nlp/tokenizers/) and make use of the
-tokenizer your model uses!
+Add this to:
+- `XXBackbone`
+- `XXTokenizer`
 
-#### Unit Tests
+[Example](https://github.com/keras-team/keras-hub/blob/master/keras_hub/src/models/distil_bert/distil_bert_backbone.py#L187-L189)
 
-The last step here is to add unit tests for the tokenizer. A dummy vocabulary is
-created, and the output of both these layers is verified including tokenization,
-detokenization, etc.
+#### Preset Tests
+The testing for presets is divided into two:
+- "Large" tests: validate smallest preset numerics
+- "Extra large" tests: loop over all presets, check for successful load/inference
 
-### Step 4: PR #3 - Add XX Presets
+#### Checkpoint Conversion Script (tools/checkpoint_conversion/convert_your_model_checkpoints.py)
```

Comment:
We should mention timm/huggingface converters. And show one of those as the primary example.

Basically, we should say that is our preferred mode of checkpoint is to go from a supported conversion source (timm, transformers) and write a built in library converted. Then the convert checkpoint tool is just a thin wrapper around this converter.

Alternately (and more advanced) would be to write a converted from another format, directly in the `tools/` script.